pegasus/sites.v3/code.org/public/private/privileges/index.haml

```diff
@@ -14,50 +14,39 @@
   messages = []
 
   def grant_permission_to_email(email, permission_name)
-    exit = false
-
-    if user = USERS.where(email:email).first
-      #user = USERS.where(email:email).first
+    user = USERS.where(email: email).first
+    if !user
+      return "#{email} is not a user"
+    end
 
-      if PERMISSIONS.where(user_id:user[:id], permission:permission_name).count > 0
-        return "#{email} already has '#{permission_name}'"
-      else
-        if permission_name == 'teacher'
-          if user[:user_type] == permission_name
-            message = "#{email} already has 'teacher'"
-          else
-            USERS.where(email: email).update(user_type: permission_name)
-            message = "#{email} granted 'teacher'"
-          end
-          exit = true
-        elsif permission_name == 'admin'
-          if user[:admin]
-            message = "#{email} already has 'admin'"
-          else
-            USERS.where(email: email).update(admin: 1)
-            message = "#{email} granted 'admin'"
-          end
-          exit = true
-        end
+    if permission_name == 'teacher'
+      if user[:user_type] == permission_name
+        return "#{email} already has 'teacher'"
+      end
+      user.update(user_type: permission_name)
+      return "#{email} granted 'teacher'"
+    elsif permission_name == 'admin'
+      if user[:admin]
+        return "#{email} already has 'admin'"
+      end
+      user.update(admin: 1)
+      return "#{email} granted 'admin'"
+    end
 
-        if permission_name == 'affiliate' && user[:user_type] != 'teacher'
-          message = "Cannot grant #{email} 'affiliate' because #{email} is not a teacher"
-          exit = true
-        end
+    if permission_name == 'affiliate' && user[:user_type] != 'teacher'
+      return "Cannot grant #{email} 'affiliate' because #{email} is not a teacher"
+    end
 
-        if !exit
-          created_at = DateTime.now
-          PERMISSIONS.insert(user_id:user[:id], permission:permission_name, created_at:created_at, updated_at:created_at)
-          message = "#{email} granted '#{permission_name}'"
-        end
-        return message
-      end
-    else
-      return "#{email} is not a user"
+    if !PERMISSIONS.where(user_id: user[:id], permission: permission_name).empty?
```

Comment:
`if PERMISSIONS.where(user_id: user[:id], permission: permission_name).any?` might work to un-double-negative